src/modules/universal-discovery/universal.discovery.module.js

```diff
@@ -110,24 +118,31 @@ export default class UniversalDiscoveryModule extends Component {
         this.setState(state => Object.assign({}, state, { contentMeta, isLocationAllowed }));
     }
 
+    onBookmarkRemoved(item) {
```

Comment:
I think I don't understand the flow.

Reply:
When a user clicks on an item in the table in Bookmarks, the Content Meta Preview (CMP) is being opened. In CMP you can unbookmark a content, in such case bookmark should be immediately removed from the table in Bookmarks. So, when it comes to implementation, I pass in UDW `onBookmarkRemoved` method to CMP, which is called if a bookmark was removed. If `onBookmarkRemoved` is called, CMP is closed (`this.onItemSelect(null)`) and `onBookmarkRemoved` method is called on `BookmarksPanelComponent`, which calls `onBookmarkRemoved` method from `BookmarksComponent`, which removes the bookmark.